### DISCUSSION TITLE
embed buttons / elements within trial

### DISCUSSION THREAD

**Discussion by EceKayaMPI** [jspsych/jsPsych#720] [Q&A]
Hi, 
I have been trying to insert a button to play sounds within trial. I have found a workaround but couldn't help but wonder if I can insert other elements than images within trial (maybe in html-keyboard response type?)
Would appreciate any input or examples.

**Comment by jodeleeuw** [comment]
If you are looking to create new modes of interaction within a trial then your best bet may be to create your own plugin file. It's certainly possible to get this working with the `html-keyboard-response` plugin, but I think you'll end up doing nearly all the work needed for a custom plugin by hacking the parameters to do what you want.

There's an admittedly brief [tutorial](https://www.jspsych.org/plugins/creating-a-plugin/) in the documentation. Feel free to come back here with questions if you run into issues creating the plugin.
